### PR TITLE
Created a JavaFX app for Drifty GUI

### LINKED ISSUE #231: [FEATURE] Changing Drifty GUI to a JavaFX interface
### Description

We talked briefly about the Drifty GUI interface in #225 so I wanted to open this one up specifically for this PR that I WANT to submit, but I've run into some issues that have been difficult to figure out. I've invested way too many hours into this to just toss it out the window so I'm hoping some collaboration will help me get this interface completed.

I have a lot of questions, so I'll try to stick to what's more relevant... 

I'm new to `yt-dlp` or any of the forks from the original. It has enormous capability, but of course, leveraging all of that capability within a wrapper program can be very overwhelming. I noticed in the code, that you seem to have broken down capabilities offered to your users in generally three "categories" which are Youtube downloads, Instagram downloads, and "everything else"...

I've been focusing my efforts on just the youtube feature of the code. But when I thought it would be simple enough to just use the same method calls that you use in the Swing code which seems straightforward, but there are functions even in the Swing code that aren't quite working right, such as filename detection, etc. You also seem to rely heavily on the MessageBroker class for passing information around in the program which I think might also be connected to logging somehow. I haven't studied that area of the code in any depth because it looks a little intimidating at first glance.

I was able to get filename detection working using these command switches in yt-dlp

```Java
--write-info-json --skip-download -P /path urlLink
```
That writes a very thorough JSON file that I then use to grab the filename. And as it turns out, when your link is a list of youtube videos, it creates one JSON file for each video in the list which also includes the URL for each file, making it easy to create batch jobs from links like that.

Could you help me understand the messaging framework a little better? Whats the intent there, how should it be utilized etc.?  Also what the primary goal has been for the project given the abilities of yt-dlp.... has your scope just been youtube and Instagram mainly?

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by EasyG0ing1**
## Fixes issue

Fixes #231 

## Check List (Check all the applicable boxes) 
- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers
The discussions about this PR has been continued in our [Discord Server](https://discord.gg/DeT4jXPfkG).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -2,4 +2,4 @@ blank_issues_enabled: false
 contact_links:
   - name: Are you having any questions?
     url: https://discord.gg/DeT4jXPfkG
-    about: Feel free to ask your questions on our Discord server.
+    about: Feel free to isYes your questions on our Discord server.
```
Comment (SaptarshiSarkar12) [inline .github/ISSUE_TEMPLATE/config.yml:5]: This file change is unnecessary. Please remove this.

**Comment by SaptarshiSarkar12** [comment]
Please don't change any workflows. Its not a part of GUI.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (EasyG0ing1) [reply .github/ISSUE_TEMPLATE/config.yml:5]: Interesting, I don't recall touching that file, but I will correct it.

**Comment by EasyG0ing1** [comment]
> 

Any NON-GUI changes were completely accidental.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -46,8 +46,8 @@ jobs:
         cache: maven
     - name: Build with Maven for ${{ matrix.os }}
       run: mvn compile
-    - name: Package Drifty CLI for ${{ matrix.os }} with JavaPackager Maven plugin
-      run: mvn javapackager:package@bundling-Drifty-CLI-for-${{ matrix.os }}
+    - name: Package Drifty CLIString for ${{ matrix.os }} with JavaPackager Maven plugin
```
Comment (SaptarshiSarkar12) [inline .github/workflows/build.yml:49]: @EasyG0ing1 Please don't make any changes in workflow file.

Comment (EasyG0ing1) [reply .github/workflows/build.yml:49]: This was not intentional. It must have happened accidentally with a global search and replacement within IntelliJ.

Comment (SaptarshiSarkar12) [reply .github/workflows/build.yml]: Ok

**Comment by SaptarshiSarkar12** [comment]
@EasyG0ing1 Due to some GraalVM and GluonFx native image generation restrictions, I am going to make some changes in some files under GUI and Backend packages along with a massive change in the pom.xml file PERMANENTLY replacing JavaPackager with GraalVM and GluonFX. So, I shall push those changes once you have all your local changes pushed to avoid merge conflicts. Even the workflows are going to change.
Let me know if you are comfortable with this.
I would also like to request you to let me know when I can push those changes, as soon as possible.

**Comment by SaptarshiSarkar12** [comment]
@EasyG0ing1 I have pushed all the changes that I have made. Please **do not revert any of those changes** (if needed, *please inform me*).
**Workflow** and **pom files** ***are subjected to change in the future***, once ***[some issues with building on macOS gets resolved](https://github.com/gluonhq/gluonfx-maven-plugin/issues/477#issuecomment-1675717627)***.

**Comment by SaptarshiSarkar12** [comment]
@EasyG0ing1 The build workflow changes have successfully been made. You can continue with the GUI development task discussed on [our Discord server](https://discord.gg/DeT4jXPfkG).
 Are you facing any problem? You seem to be inactive for three days.

**Comment by SaptarshiSarkar12** [comment]
@EasyG0ing1 Please respond.

**Comment by SaptarshiSarkar12** [comment]
@EasyG0ing1 I have made some changes in the GUI and CLI packages as under.
- GUI
    - Removed direct call to methods like **`setLinkOutput()`**, **`setFilenameOutput()`**, **`setDirOutput()`** and **`setDownloadOutput()`**. These methods will now be called by **MessageBroker** class.
    - Fixed the bug of **Validating link...** which never stops, by adding the **isUrlValid** call under the respective else branch.
    - Added a line of code to log the starting of the Drifty GUI application.
- CLI
    - Added support for retrieving filenames for YouTube and Instagram videos.
    - Fixed a file downloading issue in Drifty CLI
    - Added **`initializeEnvironment()`** method to initialize all the required parameters.
- Some Common changes
    - Moved **MessageBroker** instance to be present only in environment class. It will be set and accessed by **`setMessageBroker()`** and **`getMessageBroker()`** methods. So, some parts of both the CLI and GUI has got changed.
    - Changed **`findFilenameInLink()`** to provide filenames for both YouTube, Instagram and normal file download links (like github file links, iso download links, etc.).
    - The **`isUrlValid`** would now handle the exceptions and returns boolean true/false according to the validity of the link.

**Comment by SaptarshiSarkar12** [comment]
I have made those changes just to extract the common parts out from both CLI and GUI as much as possible.

**Comment by SaptarshiSarkar12** [comment]
@EasyG0ing1 You seem to be inactive for more than 4 days. When will you be free to work on this issue? 
Please note that #217 is waiting for this PR to get merged. So, we need to have this PR merged with all (or at least most) bugs resolved, within one week if possible. Let me know if you are comfortable with this.

**Review by SaptarshiSarkar12 — APPROVED** [review]
**Looks good to merge 👍.** 
@EasyG0ing1 **Thanks for contributing 🚀 🚀.**

**Pull request merged**